### PR TITLE
Add Elasticsearch Vector Store Connector

### PR DESCRIPTION
**Pull request by flobernd**
Adds the Elasticsearch Vector Store Connector documentation.

cc @westey-m

### CODE REVIEW

**Comment by flobernd** [comment]
@microsoft-github-policy-service agree company="Elastic"

File: semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      0.9,
+      0.1,
+      0.1,
+      0.1
+    ]
+  }
+}
+```
+
+::: zone-end
```
Comment (westey-m) [inline semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:202]: To fix the warnings, you'll have to add the python and java sections as well.  What we are typically doing is adding coming soon for those or whatever the appropriate message may be, e.g.

```
::: zone pivot="programming-language-python"

## Coming soon

More info coming soon.

::: zone-end
::: zone pivot="programming-language-java"

## Coming soon

More info coming soon.

::: zone-end
```

File: semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+|-----------------------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| Collection maps to                | Elasticsearch index                                                                                                              |
+| Supported key property types      | string                                                                                                                           |
+| Supported data property types     | All types that are supported by System.Text.Json (etiher built-in or by using a custom converter)                                |
+| Supported vector property types   | <ul><li>ReadOnlyMemory\<float\></li><li>IEnumerable\<float\></li></ul>                                                           |
+| Supported index types             | <ul><li>HNSW (32, 8, or 4 bit)</li><li>FLAT (32, 8, or 4 bit)</li></ul>                                                          |
+| Supported distance functions      | <ul><li>CosineSimilarity</li><li>DotProductSimilarity</li><li>EuclideanDistance</li><li>MaxInnerProduct</li></ul>                |
```
Comment (westey-m) [inline semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:27]: With the other connectors the names provided here are the same as what is supposed as a parameter.  E.g. to use `EuclideanDistance` you can pass "EuclideanDistance".  Is that the same here as well, for e.g. `MaxInnerProduct`?

**Comment by westey-m** [comment]
@flobernd, thanks!  Just added two small comments, but otherwise looks great!

File: semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md (hunk 2 above)
Comment (flobernd) [reply semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:27]: Thanks for the review!

Yes, that's the case here as well. The similarity measure / distance function is modelled as an enum in the underlaying Elasticsearch client library and I made sure to map the string constants defined in the SK abstractions accordingly. For `MaxInnerProduct`, which is specific to Elasticsearch, I've adapted the existing naming convention and made sure to as well map it to the corresponding internal enum member.

File: semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md (hunk 1 above)
Comment (flobernd) [reply semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:202]: The invalid author warnings are probably not as easy to fix. Should I just use your name or how do we handle these?

Comment (westey-m) [reply semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:202]: Sure, you can use the following, thanks!
```
ms.author: westey
```

File: semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md (hunk 2 above)
Comment (westey-m) [reply semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/elasticsearch-connector.md:27]: Perfect, sounds good.  Thanks!

**Review by westey-m — APPROVED** [review]

**Pull request merged**